### ISSUE TITLE
Error when installing `requests==2.26.0`

### ISSUE THREAD

**Issue by rhoriguchi** [DavHau/mach-nix#342]
Error when installing `requests==2.26.0`, for some reason it will just remove all dependencies, and then it will fail because dependencies are missing. Same behavior when just installing `requests==2.26.0`.

setup.py
```py
from setuptools import setup, find_packages

setup(
...
    install_requires=[
        'beautifulsoup4==4.9.3',
        'PyYAML==5.4.1',
        'requests==2.26.0'
    ],
...
)
```

Log output of `nix-build`
```bash
trace:
 automatically detected requirements of APPLICATION:VERSION:
beautifulsoup4==4.9.3
PyYAML==5.4.1
requests==2.26.0
...
### Resolved Dependencies ###

beautifulsoup4 - 4.9.3 - wheel
└── soupsieve - 2.2.1 - wheel

pyyaml - 5.4.1 - wheel

requests - 2.26.0 - nixpkgs

Total number of python modules: 4
Removed circular dependencies: 0

trace: removing dependency python3.9-brotlicffi-1.0.9.2 from requests
trace: removing dependency python3.9-certifi-2021.05.30 from requests
trace: removing dependency python3.9-charset-normalizer-2.0.1 from requests
trace: removing dependency python3.9-chardet-4.0.0 from requests
trace: removing dependency python3.9-idna-3.2 from requests
trace: removing dependency python3.9-urllib3-1.26.6 from requests
trace: removing dependency python3.9-brotlicffi-1.0.9.2 from requests
trace: removing dependency python3.9-charset-normalizer-2.0.1 from requests
...
ERROR: Could not find a version that satisfies the requirement certifi>=2017.4.17 (from requests) (from versions: none)
ERROR: No matching distribution found for certifi>=2017.4.17
...
```

**Comment by rhoriguchi** [comment]
Possible related to #330

**Comment by DavHau** [comment]
fixed in version 3.4.0
